### ISSUE TITLE
SPIKE Drupal Investigation for Image Uploader

### ISSUE THREAD

**Issue by Agile6MSkinner** [department-of-veterans-affairs/va.gov-cms#18710]
## Description
The IIR team has been doing some research and design work to figure out a way to improve the image upload process for editors.

David Pearl (IIR UX Designer) has created wireframes that we need to investigate the technical feasibility on. There could be custom Drupal modules that give us exactly what we need for capabilities, there could be configurations we could leverage, etc. The IIR just needs to know what is or isn’t possible within the current system. We want to avoid custom code if possible.

## Notes/Resources:
David has created wireframes that outline today’s current image uploading process vs. what we would would like in a future state (labeled "For Consideration"). 

[Figma Designs](https://www.figma.com/design/OjptXN3dzYqJHDQb8zV2eb/CMS%2FVAMC-Research?node-id=69-809&t=Zu8fePLf44ingGNW-1)

## Acceptance Criteria:
1. Make comments in the Figma design for what is possible vs not possible within existing Drupal modules/configurations.

```[tasklist]
### Tasks List
- [x] Meet with David and Megan so they can walk through the Figma wireframes - Megan will be setting up that meeting
- [x] Investigate feasibility of proposed wireframe
- [x] Make comments in Figma on proposed wireframe based on Drupal findings
- [x] Schedule a quick follow up meeting after investigation with David and Megan to review findings
```

**Comment by aklausmeier** [comment]
Meeting with IIR team, David and Megan, for a walk through of the wireframes (task 1) is scheduled for Monday, July 29.

**Comment by omahane** [comment]
Engineers from Facilities met with IIR on 29 July 2024. David Paul went over his UX recommendations. The engineers agreed to provide comments on the Figma design. Engineers then met separately immediately afterward and also met again with David on 30 July to confirm that there should be no front-end impact and another cropping tool was suggested (Focal Point) to be tested against the current one (Image Widget Crop)

He also told us that the priority UX changes were
- Removing the "Description" field from the image upload form
- Fixing the crop UX, which currently shows all the crops, even though a given field will only use a single crop.

My recommendations are as follows:
- Patch core to select a form display for the Media Library widget

- Create 4 Form displays for image for each of the following:
  - 7:2 crop
  - 3:2 crop
  - square crop
  - full-width, no upscaling, no crop
- Remove the "Description" field from the form (either in all form displays or the custom ones with the specific styles/crops mentioned above)

- Update the necessary content types and fields, per this spreadsheet:
  - https://docs.google.com/spreadsheets/d/150p32AtPODemMvcckTTotZ4DWVIPdzfF907CPwlBTXA/edit?gid=0#gid=0

- Confirm no front-end regressions


Items to not address:
- Creating separate workflows for uploading vs selecting an image
- Anything with alt text as there's an [epic](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/17443) to address it 
- Setting the section by default (as it already is winnowed to only the Sections the user can access)

Items already addressed:
- Image file names prepopulate in the Name field
- Drag and drop functionality is present

Items that require more investigation
- Getting image previews with the appropriate crop in node:edit and node:view
- Allowing the user to edit an image from the node on which it appears, rather than just remove it, which is the only current option

**Comment by omahane** [comment]
@Agile6MSkinner @jilladams

We had a follow-up meeting on 31 July, where we went over my findings and recommendation and also heard from @edmund-dunn about his own suggestions. 

David and Megan said that their next step was to do UX research on these recommendations, using an interactive prototype. For that purpose, they asked the Drupal engineers to identify any items that would not be likely candidates for changes, due to their complexity or their status as core functionality. I have made those notes on the Figma document.

@edmund-dunn said that he could try to set up an environment where he would make the image preview match the chosen image style (such as an image cropped at 7:2).

I believe that we have done the acceptance criteria of this ticket. The branch with the work I did to test this is https://github.com/department-of-veterans-affairs/va.gov-cms/tree/VACMS-18710-spike-media-library
